flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/codegen/calls/CallGenerator.scala

```diff
@@ -43,11 +43,16 @@ object CallGenerator {
     val nullTerm = newName("isNull")
     val resultTypeTerm = primitiveTypeTermForTypeInfo(returnType)
     val defaultValue = primitiveDefaultValue(returnType)
+    val nullCheckTerms = if(operands.size > 0) {
+      operands.map(_.nullTerm).mkString(" || ")
+    } else {
+      nullCheck + ""
+    }
```

Comment:
It seems that when operands.size == 0  and nullCheck is enable, the `$nullTerm` will always be true, which means `$resultTerm` always be the default value.  That may be wrong.